src/operators.jl

```diff
@@ -210,12 +286,12 @@ _sizehint_expr!(q, n) = nothing
 #############################################################################
 
 # TODO: specialize sum for Dict and JuMPArray of JuMP objects?
-Base.sum(j::Array{<:AbstractVariableRef}) = GenericAffExpr(vec(j), ones(length(j)), 0.0)
+Base.sum(vars::Array{<:AbstractVariableRef}) = GenericAffExpr(0.0, [v => 1.0 for v in vars])
```

Comment:
Isn't more inference-friendly to use `map` instead of a generator ? I always try to avoid them because `eltype(v => -1.0 for v in vars)` is `Any`

Reply:
I added a test with `@inferred` and this passes.